### ISSUE TITLE
Master Template / UI Guide

### ISSUE THREAD

**Issue by mdfrederick** [Masterworks/Clarity-Components#4]
Also known as the UI Guide - https://projects.invisionapp.com/share/GCL1U3CKB45#/screens/298862244_Elements

**Comment by mdfrederick** [comment]
@Chitradip-Capitalnumbers Can you edit/comment with some detail around what this is? I'm guessing its the header/menu/container/footer elements which will be the basis for all/most of the pages?

**Comment by Chitradip-Capitalnumbers** [comment]
@mdfrederick 
Yes, you are right. Actually, this is just the full view of all component related to Clarity app. This is needed to avoid the internal class conflict. If we do so then in the time of development a Vue Js developer can easily take any type of component without any dependency.

**Comment by Chitradip-Capitalnumbers** [comment]
@mdfrederick 
I have completed Master template. Also pushed that on the GitHub. You can check that. Now you can close this issue.

**Comment by danielgolden** [comment]
@Chitradip-Capitalnumbers  Thanks for your work on this, I've reviewed it and will provide feedback below:

- [ ] Checkbox labels shouldn’t be uppercase and shouldn’t have any extra letter-spacing.
### Dropdown Element 
- [ ] The text in `.dropdown-content li` is vertically off-center.
- [ ] When an `<li>` in the sub-nav section (`.level-2-dropdown`) of `dropdown-content` is hovered over there’s a 1 pixel shift happening that shouldn’t:
![kapture 2018-06-21 at 6 24 30](https://user-images.githubusercontent.com/812989/41714281-c9ab4c80-751d-11e8-8b90-7e58321fe161.gif)

### Text & Select inputs
- [ ] The padding-left on the “has no input/value” state is not correct on the standard-size and small inputs. Please take the measurements directly from Figma for this.
- [ ] The labels for these inputs is also a bit too large when the input has focus. It may be helpful to. Please also take measurements directly from Figma for this.
- [ ] The dropdown arrow for the standard size select in off-center vertically.

**Comment by danielgolden** [comment]
@Chitradip-Capitalnumbers I should also mention that a big chunk of these issues (if not most of them) are left over from our first build of these components, so I do recognize that we created the bugs, not you. That said I appreciate your resolving them.

**Comment by danielgolden** [comment]
@Chitradip-Capitalnumbers Most of these issues have been resolved, but several still remain. Here are the issue that remain: 
- The select input labels have letter spacing even when their inputs are not focused (They should not have letter spacing until the input gets focus)
- The label for the standard sized input is still incorreclty positioned when it's input has focus (it's too far to the right) and it's still not set in uppercase.
- The text of the tag in the grid view item is off-center vertically

**Comment by mdfrederick** [comment]
@Chitradip-Capitalnumbers Please structure this page more like the Invision screen instead of just a list of components. This isn't just to avoid internal class conflict, this is a resource to be developed and re-used by anyone working on the project.

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden 
Hello,
I have removed the bootstrap grid from the project and implemented my own custom style there.
I have also try to match the spacing issue , but as the design element and the web element both
are from different media you may find 1px 2px difference due to browser rending techniques. Today I have also work on select-client.html page, please check and let me know your view.

**Comment by danielgolden** [comment]
@Chitradip-Capitalnumbers I think this grid will work well. Thank you.

**Comment by danielgolden** [comment]
> @Chitradip-Capitalnumbers Please structure this page more like the Invision screen instead of just a list of components. This isn't just to avoid internal class conflict, this is a resource to be developed and re-used by anyone working on the project.

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden 

I have made some more progress on this page. I have pushed the code. Please review my progress on this page :- ui-guide-v2.html

Please check and let me know your thoughts.

Thanks!

**Comment by Chitradip-Capitalnumbers** [comment]
@danielgolden 
I have completed the UI guide page.
Please check and let me know your feedback.

Thanks!

**Comment by danielgolden** [comment]
The form inputs do not match what's in figma. Please pay close attention to what is in figma and match it in the UI guide.
